api-server/src/roster/views.py

```diff
@@ -345,4 +353,134 @@ def get_queryset(self):
+                - Upload those to Blob
+        """
+        if not request.user.is_staff:
+            return Response({'detail': ['Not allowed']},
+                            status=status.HTTP_403_FORBIDDEN)
+
+        aviators = Aviator.objects.all()
+        if not aviators:
+            return Response({'detail': ['No Aviators found']},
+                            status=status.HTTP_404_NOT_FOUND)
+
+        for aviator in aviators:
+            squadron = aviator.squadron.designation
+
+            # This is needed as DCS needs a very specific name for each airframe
+            airframe_label = aviator.squadron.air_frame.dcs_livery_label
+
+            # Get liveries from DB
+            squadron_livery = Livery.objects.filter(squadron__designation = squadron)\
+                    .filter(squadron__air_frame__name = aviator.squadron.air_frame.name)\
+                    .filter(position_code = aviator.position_code)
+            if not squadron_livery:
+                continue
+
+            squadron_livery = squadron_livery[0]
+            # Get all the livery's skins
+            skins = squadron_livery.skins.all()
+            
+            # Create custom DDS file for aviator and save in blob storage
+            for skin in skins:
+                file_name = os.path.basename(skin.dds_file.name)
+                dds_path = f"livery/{airframe_label}/{squadron} {aviator.callsign}/{file_name}"
+                self.create_aviator_dds(skin, aviator, dds_path)
+
+            lua_path = f"livery/{airframe_label}/{squadron} {aviator.callsign}/description.lua"
+            lua_sections = squadron_livery.lua_sections.all()
+            self.create_aviator_lua(aviator,lua_sections, lua_path)
+           
+        return Response({'detail': "Complete"}, status=status.HTTP_201_CREATED)
+
+    def create_aviator_lua(self, aviator, lua_sections, lua_path):
+
+        lua_text = "livery = {\n"        
+        for lua_section in lua_sections:
+            lua_text += f"\n{lua_section.text}\n"
+        lua_text += f"\n}}\nname = \"{aviator.squadron.designation} {aviator.callsign}\""
+
+        self.block_blob_service.create_blob_from_text(self.azure_container, lua_path, lua_text)
+            
+
+    def create_aviator_dds(self, skin, aviator, blob_path):
+
+        with Image(file=skin.dds_file.open(mode='rb')) as img:
+            for prop in skin.json_description:
+                with Image(width=prop["img_size"]["width"], height=prop["img_size"]["height"]) as tmp_image:
+                    draw = Drawing()
+
+                    if "font" in prop:
+                        draw.font = prop["font"]
+
+                    if "font_size" in prop:
+                        draw.font_size = prop["font_size"]
+
+                    if "font_opacity" in prop:
+                        draw.fill_opacity = prop["font_opacity"]
+
+                    if "font_alignment" in prop:
+                        draw.text_alignment = prop["font_alignment"]
```

Comment:
it might be more concise to do something like this

```suggestion
                    draw = Drawing(**prop)
```

this will be a keyword unpacking of the props object passed to the Drawing object.

Reply:
👍

Reply:
@kruzaavn I am afraid Drawing() complains about the extra arguments that are passed to it.
As we need some extra arguments for the logic, I'm going to leave this as is. But I agree that unpacking looked better.